libc-test/build.rs

```diff
@@ -1768,51 +2470,184 @@ fn test_dragonflybsd(target: &str) {
     cfg.skip_type(move |ty| {
         match ty {
             // sighandler_t is crazy across platforms
+            // FIXME: is this necessary?
             "sighandler_t" => true,
 
+            // FIXME: these types have a p_type member but some header
+            // has a define p_type __p_type macro that breaks their tests:
+            "Elf64_Phdr" | "Elf32_Phdr" => true,
```

Comment:
Not in scope of this PR:
I suppose it's from `resolv.h`, maybe headers should be tested separately so they don't conflict with each other?

Reply:
It is from `resolv.h`, the problem is, why is this issue triggering now, and was not triggering before.

Reply:
`Resolv.h` doesn't change a lot: https://sourceware.org/git/?p=glibc.git;a=history;f=resolv/resolv.h;hb=7ac6fad987c28dc5e0e7964084260f163179be68

Maybe doing diff could show something.

Reply:
I worry that this is a header file ordering issue.

Reply:
I've just split the test for now.